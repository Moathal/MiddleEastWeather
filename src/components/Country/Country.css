.countryContainer {
  display: flex;
  align-items: center;
  width: calc(50vw - 15.5px);
  height: calc(50vw - 10px);
  box-shadow: 0 0 6px rgb(0, 29, 56);
  justify-content: space-between;
  padding: 1px;
  margin: 3px;
  text-decoration: none;
}

img.left {
  max-width: 25vw;
  max-height: 25vw;
  margin-left: 3px;
}

.right {
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-end;
  margin-right: 5px;
  margin-bottom: 3px;
}

.goto {
  margin-top: 5px;
  height: 30px;
  width: 30px;
}

.name {
  color: lightgray;
}
