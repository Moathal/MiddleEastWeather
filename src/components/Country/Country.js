import React from 'react';
import { Link } from 'react-router-dom';
import goto from '../../assets/group55.png';
import './Country.css';

function Country({ name, id, img }) {
  return (
    <div>
      <Link className="countryContainer" to={`/details/${id}`}>
        <img className="left" src={img} alt="country's Map" />
        <div className="right">
          <img className="goto" src={goto} alt="goto icon" />
          <p className="name">{name}</p>
        </div>
      </Link>
    </div>
  );
}

export default Country;
