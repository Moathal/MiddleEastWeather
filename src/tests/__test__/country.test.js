import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import Country from '../../components/Country/Country';

it('renders Country correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <Country />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
