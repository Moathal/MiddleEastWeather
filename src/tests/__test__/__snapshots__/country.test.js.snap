// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Country correctly 1`] = `
<div>
  <a
    className="countryContainer"
    href="/details/undefined"
    onClick={[Function]}
  >
    <img
      alt="country's Map"
      className="left"
    />
    <div
      className="right"
    >
      <img
        alt="goto icon"
        className="goto"
        src="group55.png"
      />
      <p
        className="name"
      />
    </div>
  </a>
</div>
`;
